engine/src/main/java/com/datatorrent/stram/StreamingContainerManager.java

```diff
@@ -1258,6 +1294,10 @@ public StreamingContainerAgent assignContainer(ContainerResource resource, InetS
       return null;
     }
 
+    //get groupid if container is deployed as part of redeployemnt process
+    if (redeployRequests.containsKey(container.getExternalId())) {
```

Comment:
Not sure I understand this logic. 2 things either not clear or correct: 1) if the externalContainerId (YarnId) is removed below on line 1305, shouldn't you also remove the entry from redeployRequests for the externalId? Why is it still kept around? 2) as a result of not removing, will you by mistake be carrying forward very old groupId from when the removed container was restarted? e.g. container 1 was started 5 hours ago and is being redeployed now because of an error, wouldn't you be using the old group id fro 5 hours ago?